cloud/services/container/clusters/kubeconfig.go

```diff
@@ -108,18 +108,15 @@ func (s *Service) createUserKubeconfigSecret(ctx context.Context, cluster *conta
 		return fmt.Errorf("creating base kubeconfig: %w", err)
 	}
 
-	authProviderConfig := &api.AuthProviderConfig{
-		Name: "gcp",
-		Config: map[string]string{
-			"cmd-args":   "config config-helper --format=json",
-			"cmd-path":   "gcloud",
-			"expiry-key": "'{.credential.token_expiry}'",
-			"token-key":  "'{.credential.access_token}'",
-		},
+	execConfig := &api.ExecConfig{
+		APIVersion:         "client.authentication.k8s.io/v1beta1",
+		Command:            "gke-gcloud-auth-plugin",
```

Comment:
@Shaad7 - should we only be using this from v1.26?

Reply:
Yes. The auth changed in GKE version 1.26

Reply:
@richardcase Should I add a condition to check the GKE version and generate kubeconfig according to the version ?

Reply:
If it's version specific, then I think we will need to. wdyt?

Reply:
It is not version specific. The update works on all the versions and the old one does not. So, I think we are good to go with the PR.

Reply:
Thanks for the clarification, and sorry its taken so long to get back to you.